code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
I dont think this should be the case as "Stage" is never cached due to the logic in enabled().
But I'll confirm.

As for the generating two sessions problem, I don't what the implications of this problem are.
Would it be better to call Versioned::choose_site_stage() earlier before Session::clear_all()?